rust/tw_keypair/src/secp256k1/private.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright © 2017-2023 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+use crate::secp256k1::public::PublicKey;
+use crate::secp256k1::signature::Signature;
+use crate::traits::SigningKeyTrait;
+use crate::Error;
+use k256::ecdsa::{SigningKey, VerifyingKey};
+use k256::elliptic_curve::sec1::ToEncodedPoint;
+use k256::{AffinePoint, ProjectivePoint};
+use tw_hash::H256;
+use tw_utils::traits::ToBytesVec;
+
+pub struct PrivateKey {
```

Comment:
I'd document that this must be created via `TWPrivateKey`. Same for starkex.

Reply:
In my personal opinion, `TWPrivateKey` should be used if the end-user needs to sign hashes/messages using multiple curves with the same secret key. So it's more for the Wallet Core end user.
But if there is a need to work with only one curve, it's better to use the specific curve implementation directly as it's safer.